crates/voicevox_core/src/user_dict/dict.rs

```diff
@@ -0,0 +1,103 @@
+use derive_getters::Getters;
+use std::io::Read;
+use std::{collections::HashMap, fs::File};
+use uuid::Uuid;
+
+use super::word::*;
+use crate::{Error, Result};
+
+/// ユーザー辞書。
+#[derive(Clone, Debug, Getters)]
+pub struct UserDict {
+    store_path: String,
+    words: HashMap<String, UserDictWord>,
+}
+
+impl UserDict {
+    /// ユーザー辞書をロードする。
+    /// ファイルが存在しない場合は空の辞書を作成する。
+    /// ファイルが存在する場合は、その内容を読み込む。ファイルの中身が無い場合は空の辞書を作成する。
+    /// ファイルの内容が不正な場合はエラーを返す。
+    pub fn new(store_path: &str) -> Result<Self> {
+        if std::path::Path::new(store_path).exists() {
+            let mut store_file =
+                File::open(store_path).map_err(|e| Error::UserDictRead(e.to_string()))?;
+            let mut content = String::new();
+            store_file
+                .read_to_string(&mut content)
+                .map_err(|e| Error::UserDictRead(e.to_string()))?;
+
+            let words: HashMap<String, UserDictWord> = if content.is_empty() {
+                HashMap::new()
+            } else {
+                serde_json::from_str(&content[..])
+                    .map_err(|e| Error::UserDictRead(e.to_string()))?
+            };
+            Ok(Self {
+                store_path: store_path.to_string(),
+                words,
+            })
+        } else {
+            let dict = Self {
+                store_path: store_path.to_string(),
+                words: HashMap::new(),
+            };
+            dict.save()?;
+            Ok(dict)
+        }
+    }
+
+    /// ユーザー辞書に単語を追加する。
+    pub fn add_word(&mut self, word: UserDictWord) -> Result<String> {
```

Comment:
保存することも想定していることを触れていいかも。
```suggestion
    /// ユーザー辞書に単語を追加して保存する。
    pub fn add_word(&mut self, word: UserDictWord) -> Result<String> {
```